app_basics/lab1.py

```diff
@@ -0,0 +1,107 @@
+"""
+Create a cli tool that will keep running, waiting for user input, evaluating and printing.
+
+As a user, I want to be able to input a file path(hint, see os.path), apply head/tail on the file content,
+    and see the output  (for example first 3 lines)
+As a user, I want to be able to input a csv line, apply head/tail on the line content,
+    and see the output(for example first 3 columns)
+As a user, I want to be able to exit with a special command (for example '//exit')
+
+As a user, I want to be able to input anything else and see only the head/tail
+As a user, I want the program to output the accumulated first word of every line
+
+The tool should print(meaningfully) the exact date and time in which it started, and in which it ended.
+
+save current user input, while input is not exact match '//exit', keep running... (while loop with condition)
+
+testing conditions for each line:
+ test if 'exit' cmd, if true -> somehow to exit the while loop -
+
+Tamir
+=====
+Expected user input in the following line format:
+first word - commands:
+                    breakout valid values: '//quit', '//exit'
+                    head - will be followed by a number (optional default  1)
+                    tail - will be followed by a number (optional default  1)
+                    print - display the content of the rest of the line to the console.
+Second word -   file path, only if file exist will do the head/tail command on the file.
+                If not a file, see if contains commas ',' for CSV line.
+
+Third number - optional how many lines, default 3
+"""
+# SEND_REQUEST_CMD
+import os
+from datetime import datetime
+import commands.validators as valid
+import commands.util as util
+from commands.constants import constants
+
+if __name__ == '__main__':
+    counter: int = 0
+    key = constants.keys()
+    history_commands = []
+    start_time = datetime.now()
+
+    while True:
```

Comment:
Hard - try to find a way to avoid keeping count with `counter`, and use `enumerate` and `iter` instead :)
(intead of `while True`)
you will need an infinite `iterator` with an `enumerate` around it :)